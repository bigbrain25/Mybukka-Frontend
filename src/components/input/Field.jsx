import React from 'react';
import PropTypes from 'prop-types';

import './input.scss';

const Input = React.forwardRef(({
  type,
  name,
  classNames,
  placeholderText,
  handleChange,
  onFocus,
  accept,
  autoComplete,
  checked,
  value,
  onBlur,
  readOnly,
}, ref) => (
  <input
    type={type}
    name={name}
    ref={ref}
    readOnly={readOnly}
    accept={accept}
    className={classNames}
    placeholder={placeholderText}
    onChange={handleChange}
    onFocus={onFocus}
    onBlur={onBlur}
    value={value}
    autoComplete={autoComplete}
    checked={checked}
  />
));

const Checkbox = ({ onChange, onFocus, onBlur, name, value, placeholder, checked, optional, classNames = 'CheckBox-field' }) => (
  <input
    className={classNames}
    onChange={onChange}
    type="checkbox"
    required={!optional}
    onFocus={onFocus}
    onBlur={onBlur}
    name={name}
    checked={checked}
    value={value}
    placeholder={placeholder}
  />
);

const Radio = ({ onChange, onFocus, onBlur, name, value, placeholder, checked, optional, classNames = 'radio' }) => (
  <input
    className={classNames}
    onChange={onChange}
    type="radio"
    required={!optional}
    onFocus={onFocus}
    onBlur={onBlur}
    name={name}
    checked={checked}
    value={value}
    placeholder={placeholder}
  />
);

export default { Checkbox, Input, Radio };

Input.defaultProps = {
  type: 'text',
  classNames: 'Secondary-Input',
  placeholderText: '',
  onFocus: () => {},
  ref: React.createRef(),
  accept: '',
  autoComplete: 'off',
  checked: false,
  value: '',
  onBlur: () => {},
};

Input.propTypes = {
  autoComplete: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  name: PropTypes.string.isRequired,
  classNames: PropTypes.string,
  placeholderText: PropTypes.string,
  handleChange: PropTypes.func.isRequired,
  onFocus: PropTypes.func.isRequired,
  accept: PropTypes.string,
  checked: PropTypes.bool,
};
