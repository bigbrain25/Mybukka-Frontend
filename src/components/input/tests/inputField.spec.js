import React from 'react';
import Field from '../Field';

describe('Input Field', () => {
  const props = {
    name: 'test',
    onFocus: jest.fn(),
    placeholderText: 'sign in',
    handleChange: jest.fn(),
    type: 'text',
    classNames: 'default-btn',
  };
  const wrapper = shallow(<Field.Input {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('triggers the onchange function when change event ocuurs', () => {
    wrapper.simulate('change');
    expect(props.handleChange).toBeCalled();
  });
});
