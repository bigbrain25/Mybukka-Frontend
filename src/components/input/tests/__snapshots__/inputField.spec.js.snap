// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Field renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputField
    classNames="default-btn"
    defaultValue=""
    handleChange={[MockFunction]}
    handleFocus={[MockFunction]}
    name="test"
    placeholderText="sign in"
    type="text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "default-btn",
      "defaultValue": "",
      "name": "test",
      "onChange": [MockFunction],
      "onFocus": [MockFunction],
      "placeholder": "sign in",
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "default-btn",
        "defaultValue": "",
        "name": "test",
        "onChange": [MockFunction],
        "onFocus": [MockFunction],
        "placeholder": "sign in",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
